### DISCUSSION TITLE
ArduinoOTA can't find Network.h file on vscode platformio

### DISCUSSION THREAD

**Discussion by hook2r7e** [espressif/arduino-esp32#9555] [Question - Community Help]
I am having trouble getting the ArduinoOTA to build in vscode using Platformio. I get the following error message. The ArduinoOTA.h file has the #include "Network.h" line in it. It seems that platform = https://github.com/platformio/platform-espressif32.git in the platformio.ini file is not pulling in all the library folders and files. I can go to the framework folder and only certain ArduinoOTA folders and in there. It is missing several folders including the Network folder that the Network.h file is in. Any help would be appreciated.

Processing mcu (platform: espressif32 @ 6.0.1; board: esp32cam; framework: arduino)
-----------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32cam.html
PLATFORM: Espressif 32 (6.0.1) > AI Thinker ESP32-CAM
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20006.221224 (2.0.6)
 - tool-esptoolpy @ 1.40400.0 (4.4.0)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Converting rw5.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 40 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ArduinoOTA @ 2.0.0+sha.cf44890
|-- WiFiManager @ 2.0.17+sha.e978bc0
|-- Ultrasonic @ 3.0.0+sha.821cf3a
|-- NewPing @ 1.9.7
|-- ArduinoJson @ 6.21.5
|-- ESP_DoubleResetDetector @ 1.3.2
|-- ServoESP32 @ 1.0.3
|-- FS @ 2.0.0
|-- SPIFFS @ 2.0.0
|-- WiFi @ 2.0.0
Building in release mode
Compiling .pio\build\mcu\src\rw5.ino.cpp.o
Archiving .pio\build\mcu\lib91d\libUpdate.a
Compiling .pio\build\mcu\libfe5\ArduinoOTA\ArduinoOTA.cpp.o
Archiving .pio\build\mcu\lib2cb\libDNSServer.a
Archiving .pio\build\mcu\liba44\libFS.a
In file included from .pio/libdeps/mcu/ArduinoOTA/src/ArduinoOTA.cpp:5:
.pio/libdeps/mcu/ArduinoOTA/src/ArduinoOTA.h:4:10: fatal error: Network.h: No such file or directory

*****************************************************************
* Looking for Network.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:Network.h"
* Web  > https://registry.platformio.org/search?q=header:Network.h
*
*****************************************************************

 #include "Network.h"
          ^~~~~~~~~~~
compilation terminated.
Compiling .pio\build\mcu\liba84\WebServer\WebServer.cpp.o
*** [.pio\build\mcu\libfe5\ArduinoOTA\ArduinoOTA.cpp.o] Error 1
============================================================ [FAILED] Took 10.38 seconds ============================================================
 *  The terminal process "C:\Users\rmyers\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by ddnowicki** [comment]
Hi, could you send code you use, because I don't see `#include "Network.h"` in [ArduinoOTA.h](https://github.com/esp8266/Arduino/blob/master/libraries/ArduinoOTA/ArduinoOTA.h)

**Comment by hook2r7e** [comment]
It is in this link.
https://github.com/espressif/arduino-esp32/blob/master/libraries/ArduinoOTA/src/ArduinoOTA.h

I am trying to make this project work.
https://github.com/mech-toolkit/mcu/tree/main

**Reply by sidprice** [reply]
Having the same issue.

**Reply by TrevorMare** [reply]
Yup, I also have this issue

**Comment by P-R-O-C-H-Y** [comment]
From what I see you are using Arduino-esp32 version 2.0.6 and the "Network" refactoring came for 3.0.0 versions.
For your case you need to use the example from 2.X release branch, not from master:
https://github.com/espressif/arduino-esp32/blob/release/v2.x/libraries/ArduinoOTA/src/ArduinoOTA.h

**Comment by Jason2866** [comment]
Since Arduino core 3.0.0 changed the way the precompiled Arduino librarys are provided you can't do anymore
`platform = https://github.com/platformio/platform-espressif32.git`
This may change when Platformio is supporting Arduino core 3.0.0 (which will probably not happen, more info regarding this are [here](https://github.com/platformio/platform-espressif32/issues/1225)

**Comment by SamObde** [comment]
I have used this in my platformio.ini and got around that:

build_flags = -I"C:/Users/admin/.platformio/packages/framework-arduinoespressif32/libraries/Network/src"

I have no idea why I had to do that.  For some reason the network.h is there it just couldnt find it.  It was even in the .json file.

**Comment by Jason2866** [comment]
@SamObde post your platformio.ini Most probably there is something off

**Reply by SamObde** [reply]
Here it is:

```
[env:esp32dev]
platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip
board = esp32dev
framework = arduino
lib_deps = 
	https://github.com/espressif/arduino-esp32.git
	https://github.com/pschatzmann/ESP32-A2DP
	https://github.com/pschatzmann/arduino-audio-tools.git
monitor_speed = 115200
monitor_filters = esp32_exception_decoder
build_flags = -I"C:/Users/master/.platformio/packages/framework-arduinoespressif32/libraries/Network/src"
board_build.partitions = huge_app.csv
lib_ldf_mode = chain+
```

**Comment by Jason2866** [comment]
This one has the wrong config entrys removed
```ini
[env:esp32dev]
platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip
board = esp32dev
framework = arduino
lib_deps = 
	https://github.com/pschatzmann/ESP32-A2DP.git
	https://github.com/pschatzmann/arduino-audio-tools.git
monitor_speed = 115200
monitor_filters = esp32_exception_decoder
board_build.partitions = huge_app.csv
```

**Reply by SamObde** [reply]
Thanks!
